### PR TITLE
REM-512

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
new page : cb-bank-branches

### CODE REVIEW

**Comment by RebeccaFrem02** [comment]
image
country is mandatory but disabled

**Comment by RebeccaFrem02** [comment]
city, cityDistrict, state, and country are disabled

File: src/pages/cb-bank-branches/index.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+
+    filterBy,
+    clearFilter,
+    refetch,
+
```
Comment (RebeccaFrem02) [inline src/pages/cb-bank-branches/index.js:40]: remove unused new lines

File: src/pages/cb-bank-branches/index.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  return (
+    <VertLayout>
+      <Fixed>
+        <RPBGridToolbar
+          onAdd={add}
+          maxAccess={access}
+          onApply={onApply}
+          onSearch={onSearch}
+          onClear={onClear}
+          reportName={'CABNB'}
```
Comment (RebeccaFrem02) [inline src/pages/cb-bank-branches/index.js:150]: can't the report name be a label? or there is no need for it?

File: src/pages/cb-bank-branches/forms/BankBranchesForm.js [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+                    { key: 'reference', value: 'Reference' },
+                    { key: 'name', value: 'Name' }
+                  ]}
+                  values={formik.values}
+                  valueField='recordId'
+                  displayField='name'
+                  required
+                  readOnly
+                  maxAccess={maxAccess}
+                  onChange={(event, newValue) => {
```
Comment (RebeccaFrem02) [inline src/pages/cb-bank-branches/forms/BankBranchesForm.js:214]: check if this is readonly there is no need for the onChange function

File: src/pages/cb-bank-branches/forms/BankBranchesForm.js [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+                  name='cityName'
+                  label={labels.city}
+                  readOnly={(editMode || !formik.values.countryId) && true}
+                  form={formik}
+                  secondDisplayField={false}
+                  onChange={(event, newValue) => {
+                    formik.setValues({
+                      ...formik.values,
+                      cityId: newValue?.recordId || '',
+
```
Comment (RebeccaFrem02) [inline src/pages/cb-bank-branches/forms/BankBranchesForm.js:272]: remove empty line

File: src/pages/cb-bank-branches/index.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      filterFn: fetchWithFilter
+    }
+  })
+
+  async function fetchWithFilter({ filters, pagination }) {
+    if (filters?.qry) {
+      return await getRequest({
+        extension: CashBankRepository.BankBranches.snapshot,
+        parameters: `_filter=${filters.qry}`
+      })
```
Comment (RebeccaFrem02) [inline src/pages/cb-bank-branches/index.js:56]: use catch

File: src/components/Shared/apiMappings.js [hunk 6]
```diff
@@ -139,6 +139,17 @@ export const apiMappings = {
     valueField: 'recordId',
     displayField: 'name'
   },
+  [ResourceIds.Countries]: {
+    type: COMBOBOX,
+    endpoint: SystemRepository.Country.qry, //filterNationality
```
[the remark this answers is not in the document]
Comment (Alinadimmokdad) [reply src/components/Shared/apiMappings.js:144]: its needed in this case

File: src/pages/cb-bank-branches/index.js (hunk 2 above)
Comment (Alinadimmokdad) [reply src/pages/cb-bank-branches/index.js:150]: no need for it

**Pull request merged**